File: app/src/main/java/com/bkjcb/rqapplication/check/SellCheckDetailActivity.java
package com.bkjcb.rqapplication.check;

import android.content.Context;
import android.content.Intent;
import android.support.v4.app.Fragment;

import com.bkjcb.rqapplication.check.model.ApplianceCheckContentItem;
import com.bkjcb.rqapplication.check.model.ApplianceCheckResult;
import com.bkjcb.rqapplication.check.model.CheckItem;
import com.bkjcb.rqapplication.check.retrofit.ApplianceCheckService;
import com.bkjcb.rqapplication.retrofit.NetworkApi;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by DengShuai on 2019/12/19.
 * Description :
 */
public class SellCheckDetailActivity extends ApplianceCheckDetailActivity {


    @Override
    protected void getDataFromNet() {
        disposable = NetworkApi.getService(ApplianceCheckService.class)
                .getSaleCheckItem()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<ApplianceCheckResult>() {
                    @Override
                    public void accept(ApplianceCheckResult result) throws Exception {
                        if (result.pushState == 200 && result.getDatas() != null && result.getDatas().size() > 0) {
                            saveCheckContent(result.getDatas());
                            initCheckData(result.getDatas());
                            initImageListView();
                            hideEmptyView();
                        } else {
                            getDateFail(result.pushMsg);
                        }

                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        getDateFail(throwable.getMessage());
                    }
                });
    }

    @Override
    protected Fragment createFragment(ApplianceCheckContentItem item, String id) {
        return super.createFragment(item, id);
    }

    protected static void ToActivity(Context context, CheckItem checkItem) {
        Intent intent = new Intent(context, SellCheckDetailActivity.class);
        intent.putExtra("data", checkItem);
        context.startActivity(intent);
    }

    protected static void ToActivity(Context context, long id) {
        Intent intent = new Intent(context, SellCheckDetailActivity.class);
        intent.putExtra("id", id);
        context.startActivity(intent);
    }

}

File: app/src/main/java/com/bkjcb/rqapplication/check/SellCheckDetailActivity.kt
package com.bkjcb.rqapplication.check

import android.content.Context
import android.content.Intent
import android.support.v4.app.Fragment
import com.bkjcb.rqapplication.check.model.ApplianceCheckContentItem
import com.bkjcb.rqapplication.check.model.ApplianceCheckResult
import com.bkjcb.rqapplication.check.model.CheckItem
import com.bkjcb.rqapplication.check.retrofit.ApplianceCheckService
import com.bkjcb.rqapplication.retrofit.NetworkApi
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.functions.Consumer
import io.reactivex.schedulers.Schedulers

/**
 * Created by DengShuai on 2019/12/19.
 * Description :
 */
class SellCheckDetailActivity : ApplianceCheckDetailActivity() {
     override fun initDataFromNet() {
            disposable = NetworkApi.getService(ApplianceCheckService::class.java)
                    .saleCheckItem()
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe({ result ->
                        if (result.pushState == 200 && result.datas != null && result.datas!!.size > 0) {
                            saveCheckContent(result.datas!!)
                            initCheckData(result.datas!!)
                            initImageListView()
                            hideEmptyView()
                        } else {
                            getDateFail(result.pushMsg)
                        }
                    }, { throwable -> getDateFail(throwable.message) })
        }

    override fun createFragment(item: ApplianceCheckContentItem?, id: String?): Fragment? {
        return super.createFragment(item, id)
    }

    companion object {
         fun toActivity(context: Context, checkItem: CheckItem?) {
            val intent = Intent(context, SellCheckDetailActivity::class.java)
            intent.putExtra("data", checkItem)
            context.startActivity(intent)
        }

         fun toActivity(context: Context, id: Long) {
            val intent = Intent(context, SellCheckDetailActivity::class.java)
            intent.putExtra("id", id)
            context.startActivity(intent)
        }
    }
}